crates/vm/src/system/memory/merkle/tests/mod.rs

```diff
@@ -224,120 +241,144 @@ fn random_test<const CHUNK: usize>(
 
 #[test]
 fn expand_test_2() {
-    random_test::<DEFAULT_CHUNK>(3, 3000, 3, 2);
+    random_test(3, 3000, 3, 2);
 }
 
 #[test]
 fn expand_test_no_accesses() {
-    let memory_dimensions = MemoryDimensions {
-        addr_space_height: 2,
-        address_height: 1,
-    };
     let mut hash_test_chip = HashTestChip::new();
+    let height = 1;
 
-    let memory = AddressMap::new(vec![
-        1 << memory_dimensions.address_height;
-        1 + (1 << memory_dimensions.addr_space_height)
-    ]);
-    let tree = MemoryNode::<DEFAULT_CHUNK, _>::tree_from_memory(
-        memory_dimensions,
-        &memory,
-        &hash_test_chip,
+    let mem_config = MemoryConfig::new(
+        1,
+        vec![
+            AddressSpaceHostConfig {
+                num_cells: 0,
+                min_block_size: 0,
+                layout: MemoryCellType::Null,
+            },
+            AddressSpaceHostConfig {
+                num_cells: CHUNK << height,
+                min_block_size: 1,
+                layout: MemoryCellType::Native { size: 4 },
+            },
+            AddressSpaceHostConfig {
+                num_cells: CHUNK << height,
+                min_block_size: 1,
+                layout: MemoryCellType::Native { size: 4 },
+            },
+        ],
+        height + 3,
+        20,
+        17,
+        32,
     );
+    let md = mem_config.memory_dimensions();
+
+    let memory = AddressMap::from_mem_config(&mem_config);
 
-    let mut chip: MemoryMerkleChip<DEFAULT_CHUNK, _> = MemoryMerkleChip::new(
-        memory_dimensions,
+    let mut chip: MemoryMerkleChip<CHUNK, _> = MemoryMerkleChip::new(
+        md,
         PermutationCheckBus::new(MEMORY_MERKLE_BUS),
         COMPRESSION_BUS,
     );
 
-    let partition = memory_to_partition(&memory);
-    chip.finalize(&tree, &partition, &mut hash_test_chip);
+    chip.finalize(&memory, &BTreeMap::new(), &hash_test_chip);
+    let trace = chip.generate_proving_ctx();
     BabyBearPoseidon2Engine::run_test_fast(
-        vec![chip.air(), Arc::new(hash_test_chip.air())],
-        vec![
-            chip.generate_air_proof_input(),
-            hash_test_chip.generate_air_proof_input(),
-        ],
+        vec![Arc::new(chip.air), Arc::new(hash_test_chip.air())],
+        vec![trace, hash_test_chip.generate_proving_ctx()],
     )
-    .expect("This should occur");
+    .expect("Empty touched memory doesn't work");
 }
 
 #[test]
 #[should_panic]
 fn expand_test_negative() {
-    let memory_dimensions = MemoryDimensions {
-        addr_space_height: 2,
-        address_height: 1,
-    };
-
     let mut hash_test_chip = HashTestChip::new();
+    let height = 1;
 
-    let memory = AddressMap::new(vec![
-        1 << memory_dimensions.address_height;
-        1 + (1 << memory_dimensions.addr_space_height)
-    ]);
-    let tree = MemoryNode::<DEFAULT_CHUNK, _>::tree_from_memory(
-        memory_dimensions,
-        &memory,
-        &hash_test_chip,
+    let mem_config = MemoryConfig::new(
```

Comment:
probably you could have used the helper constructor and then modified it a bit, but ok